pkg/render/tiers/tiers.go

```diff
@@ -138,7 +155,34 @@ func (t tiersComponent) allowTigeraClusterDNSPolicy() *v3.NetworkPolicy {
 	}
 }
 
-func createDNSIngressNamespaceSelector(namespaces ...string) string {
+func (t tiersComponent) allowTigeraNodeLocalDNSPolicy() *v3.GlobalNetworkPolicy {
+	nodeLocalDNSPolicySelector := TigeraNamespaceSelector
+
+	return &v3.GlobalNetworkPolicy{
+		TypeMeta: metav1.TypeMeta{Kind: "GlobalNetworkPolicy", APIVersion: "projectcalico.org/v3"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: NodeLocalDNSPolicyName,
+		},
+		Spec: v3.GlobalNetworkPolicySpec{
+			Order:    &networkpolicy.AfterHighPrecendenceOrder,
+			Tier:     networkpolicy.TigeraComponentTierName,
+			Selector: nodeLocalDNSPolicySelector,
+			Egress: []v3.Rule{
+				{
+					Action: v3.Allow,
+					Destination: v3.EntityRule{
+						Nets:  []string{t.cfg.KubeDNSCIDR},
```

Comment:
This is a very unusual thing to include in a policy (a service IP): I'd like to see a comment saying why it works.

Something like:

"nodeLocalDNSCache creates and listens on the kube-dns ClusterIP on each node, so we can use that IP address directly in the policy where a normal service IP wouldn't match"